src/include/duckdb/function/table_function.hpp

```diff
@@ -198,6 +201,8 @@ typedef idx_t (*table_function_get_batch_index_t)(ClientContext &context, const
 
 typedef BindInfo (*table_function_get_bind_info_t)(const optional_ptr<FunctionData> bind_data);
 
+typedef unique_ptr<MultiFileReader> (*table_function_get_multi_file_reader_t)();
```

Comment:
I changed the way MultiFileReaders are created, passing a context is problematic because I would like to be able to call this in Deserialize methods: see the `MultiFileReader::Create` invocation in the Parquet extension